.circleci/config.yml

```diff
@@ -98,13 +101,25 @@ jobs:
           name: Check PR Metadata
           command: bundle exec danger || echo "danger failed"
       - *cache_restore_rubocop
-      - run: bundle exec fastlane execute_tests
+      - run:
+          name: bundle exec fastlane execute_tests
+          command: |
+            RUBYOPT=<< parameters.ruby_opt >> bundle exec fastlane execute_tests
       - *cache_save_rubocop
+      - run:
+          name: Check compatibility with Ruby 3.0
+          command: |
+            touch ~/test-reports/ruby_warnings.txt
+            # TODO remove 'grep -v ".bundle"' once 3rd party dependencies support Ruby 3.0
+            ! cat ~/test-reports/ruby_warnings.txt | grep -v ".bundle" | grep -E "warning:\s.*(deprecated).*$" && echo "No deprecation message found."
```

Comment:
I confirmed this script works as expected in this commit. It failed when warnings are detected.
https://app.circleci.com/pipelines/github/fastlane/fastlane/2246/workflows/258deb1d-5783-4a9e-b7f8-5c173d7b2fc8